### PR TITLE
Support concat and ident

### PR DESCRIPTION
**Pull request by jeffmay**
@audax-shreyaspurohit @mingy711 @macmacbr @danstaley
- Added support for `Cypher.ident` to create bindings.
- Added support for `:+:` operator for concatenating cypher statements
- Renamed `Statement` to `CypherStatement`

### CODE REVIEW

File: core/src/main/scala/me/jeffmay/neo4j/client/cypher/scalacheck/CypherStatementGenerators.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  implicit val shrinkCypherProps: Shrink[CypherProps] = Shrink { params =>
+    implicit val shrinkIdentifier = Shrink[String](id => Shrink.shrink[String](id).filterNot(_.isEmpty))
+    Shrink.shrinkContainer2[Map, String, CypherValue].shrink(params)
+  }
+
+  implicit val shrinkCypherParams: Shrink[Map[String, CypherProps]] = Shrink { params =>
+    implicit val shrinkIdentifier = Shrink[String](id => Shrink.shrink[String](id).filterNot(_.isEmpty))
+    Shrink.shrinkContainer2[Map, String, CypherProps].shrink(params)
+  }
```
Comment (jeffmay) [inline core/src/main/scala/me/jeffmay/neo4j/client/cypher/scalacheck/CypherStatementGenerators.scala:59]: I should document this, but empty string keys are not allowed for params or props, so this prevents the `Shrink`s from creating invalid input. I wrote these to debug the issue but it turns out that the issue was with an empty map altogether, which is allowed. I'm not sure how helpful these are, but I should at least document them.

File: build.sbt [hunk 2]
```diff
@@ -48,7 +48,6 @@ lazy val commonSettings = commonRootSettings ++ Seq(
     compilerPlugin("org.psywerx.hairyfotr" %% "linter" % "0.1.12"),
     "com.typesafe.play" %% "play-json" % playVersion,  // TODO: Allow using other Json libraries
     "org.scalacheck" %% "scalacheck" % scalacheckVersion,
-    "org.scalactic" %% "scalactic" % scalatestVersion,
```
Comment (jeffmay) [inline build.sbt:51]: This was not needed, so I removed it.

File: core/src/main/scala/me/jeffmay/neo4j/client/cypher/CypherArg.scala [hunk 3]
```diff
@@ -79,18 +96,43 @@ sealed trait CypherArg {
...
   * A label to add to either a node or relationship.
   *
+  * @see <a href="http://neo4j.com/docs/stable/graphdb-neo4j.html#graphdb-neo4j-labels">Label Documentation</a>
   * @param name the name of the label (without the preceding colon ':')
   */
-final class CypherLabel private (name: String) extends CypherLiteral(s":$name")
+final class CypherLabel private (name: String) extends CypherTemplatePart(s":$name")
 object CypherLabel {
 
   val Valid = "^[a-zA-Z0-9_]+$".r
```
Comment (macmacbr) [inline core/src/main/scala/me/jeffmay/neo4j/client/cypher/CypherArg.scala:138]: HINT: could use \w for the [a-zA-Z0-9_]  ranges

**Comment by macmacbr** [comment]
LGTM

**Comment by audax-shreyaspurohit** [comment]
lgtm

File: core/src/main/scala/me/jeffmay/neo4j/client/cypher/CypherArg.scala (hunk 3 above)
Comment (jeffmay) [inline core/src/main/scala/me/jeffmay/neo4j/client/cypher/CypherArg.scala:138]: I was having trouble with that. Also, I kinda like the explicit nature of this version. Sometimes I forget if `\w` includes `-` (it doesn't)

**Pull request merged**